ios-contact-manager/ios-contact-manager/ContactManager.swift

```diff
@@ -13,26 +13,117 @@ final class ContactManager {
+        case addContact = "1"
+        case showAllContacts = "2"
+        case findContact = "3"
+        case exit = "x"
+    }
+
+    private let phonebook = Phonebook(contacts: [:])
     
     func run() {
         do {
-            IOManager.sendOutput(type: .menu, contents: StringLiteral.start)
+            IOManager.sendOutput(
+                type: .menu,
+                contents: StringLiteral.Menu.start
+            )
             let input = try IOManager.getInput()
-            let parsedInfoInput = try parse(input)
             
-            let userInfo = try UserInfo(input: parsedInfoInput)
-            IOManager.sendOutput(type: .infomation, contents: StringLiteral.infoPrint(of: userInfo))
+            switch Menu(rawValue: input) {
+            case .addContact:
+                try addContact()
+            case .showAllContacts:
+                showAllContacts()
+            case .findContact:
+                try findContact()
+            case .exit:
+                IOManager.sendOutput(
+                    type: .infomation,
+                    contents: StringLiteral.Menu.end
+                )
+                return
+            default:
+                IOManager.sendOutput(
+                    type: .infomation,
+                    contents: StringLiteral.Menu.wrong
+                )
+            }
         } catch {
-            IOManager.sendOutput(type: .error, contents: error.localizedDescription)
+            IOManager.sendOutput(
+                type: .error,
+                contents: error.localizedDescription
+            )
         }
+        run()
     }
-    
-    private func parse(_ input: String) throws -> InfoInput {
+}
+
+
+// MARK: - addContact
+
+extension ContactManager {
+    private func parse(info input: String) throws -> InfoInput {
+        let inputPattern = #"^.+\b(?<sep>( \/ )|(\/))(\b[^\s]+\b)\k<sep>(\b[^\s]+)$"#
+        
         guard input ~= inputPattern else {
             throw IOError.invalidInputFormat
         }
-        let splitedInput = input.components(separatedBy: "/").map {$0.trimmingCharacters(in: .whitespaces)}
-        return (name: splitedInput[0], age: splitedInput[1], phone: splitedInput[2])
+        let splitedInput = input.components(separatedBy: "/").map {
+            $0.trimmingCharacters(in: .whitespaces)
+        }
+        let infoInput = (name: splitedInput[0], age: splitedInput[1], phone: splitedInput[2])
+        return infoInput
+    }
+    
+    private func addContact() throws {
+        IOManager.sendOutput(
+            type: .menu,
+            contents: StringLiteral.Contact.add
+        )
+        let input = try IOManager.getInput()
+        let parsedInfoInput = try parse(info: input)
+        let userInfo = try UserInfo(input: parsedInfoInput)
+        let inserted = phonebook.add(contact: userInfo)
+        let description = inserted ? StringLiteral.Contact.added( userInfo.addedDescription) : StringLiteral.Contact.exist
```

Comment:
p5;

불필요한 띄어쓰기가 있는 것으로 보여요 👀
```suggestion
        let description = inserted ? StringLiteral.Contact.added(userInfo.addedDescription) : StringLiteral.Contact.exist
```

Reply:
vivi... 꼼꼼히 봐주셔서 고마워요🥹 감동... 수정할게요!